modin/pandas/dataframe.py

```diff
@@ -123,7 +125,10 @@ def __init__(self, data=None, index=None, columns=None, dtype=None,
             if block_partitions is not None:
                 axis = 0
                 # put in numpy array here to make accesses easier since it's 2D
-                self._block_partitions = np.array(block_partitions)
+                if not isinstance(block_partitions, np.ndarray):
```

Comment:
Can this check be done in `_fix_blocks_dimensions`?

Reply:
Yes!